include/pmacc/nvidia/atomic.hpp

```diff
@@ -150,19 +152,21 @@ template<typename T, typename T_Acc, typename T_Hierarchy>
 HDINLINE
 T atomicAllInc(const T_Acc& acc, T *ptr, const T_Hierarchy& hierarchy)
 {
-    return detail::AtomicAllInc<T, (PMACC_CUDA_ARCH >= 300) >()(acc, ptr, hierarchy);
+    return detail::AtomicAllInc<T, (PMACC_CUDA_ARCH >= 300 || BOOST_COMP_HIP) >()(acc, ptr, hierarchy);
 }
 
 template<typename T>
 HDINLINE
 T atomicAllInc(T *ptr)
 {
-#ifdef __CUDA_ARCH__
-    return atomicAllInc(alpaka::atomic::AtomicUniformCudaHipBuiltIn(), ptr, ::alpaka::hierarchy::Grids());
-#else
-    // assume that we can use the standard library atomics if we are not on gpu
-    return atomicAllInc(alpaka::atomic::AtomicStdLibLock<16>(), ptr, ::alpaka::hierarchy::Grids());
-#endif
+    /* Dirty hack to call a alpaka accelerator based function.
+     * Member of  the fakeAcc will be uninitialized and should be not accessed.
+     *
+     * The id provider for particles is the only code where atomicAllInc without an accelerator is used.
+     * @todo remove the unsafe faked accelerator
+     */
+    pmacc::memory::Array<cupla::AccThreadSeq, 1> fakeAcc;
+    return atomicAllInc(fakeAcc[0], ptr, ::alpaka::hierarchy::Grids());
```

Comment:
@jkelling This is the new way how to handle the unknown accelerator in this function. You can remove your workaround with the pre-compiler if cascade in your SPEC branch after you rebased against this changes.